### DISCUSSION TITLE
Signed Messages...

### DISCUSSION THREAD

**Discussion by SireWilliamson** [btcpayserver/btcpayserver#5535] [Technical Support] [answered]
I need to sign a message for a crypto exchange, to verify my ownership of the wallet. 
Is it possible to do this from BTCPay?

**Comment by mp21-org** [comment]
Check out the AOPP Plugin. Should be what you are looking for:

> Allows you to support the AOPP protocol in invoices to allow customers to bypass stupid KYC rules.

**Reply by SireWilliamson** [reply]
AOPP has no documentation. I'm not sure how I would use it to do this. What does the plugin provide? Once installed, how do I use it sign a message?
I'm a little weary about installing a plugin, I know nothing about.

**Reply by mp21-org** [reply, marked as answer]
The plugin is from core contributor @Kukks - so it's definitly trusted.
After installing the plugin, you need to enable it:
![grafik](https://github.com/btcpayserver/btcpayserver/assets/1374810/f0888c45-1ee0-4c69-a6f1-5138211f3bff)

The plugin creates another tab in classic checkout:

![grafik](https://github.com/btcpayserver/btcpayserver/assets/1374810/025df306-3ddd-4ac5-a7b6-83c76cf9b953)

It seems that it's not available in the new checkout!

Not sure if your exchange adheres to the AOPP protocol and gives you a URL. The exchanges I encountered from switzerland usually just give you a message that you should sign with the private key belonging to the bitcoin address that you are trying to deposit to.

Also note that it will almost certain only work with hot wallets, because it needs access to the private keys of the wallet.

If it's not the solution to your problem: You might be better of importing the wallet via hw wallet or seed into electrum and do the message signing from there.

**Reply by SireWilliamson** [reply]
Ya, I need to make a message, sign it, and send the address to the exchange, so I don't think this will work, but thanks for the details though. 
I guess I'll have to use electrum.
I'll mark this as the answer though, as you have confirmed for me that this is not a native feature of BTCPay.
